jvb/src/main/kotlin/org/jitsi/videobridge/cc/av1/Av1DDAdaptiveSourceProjectionContext.kt

```diff
@@ -0,0 +1,648 @@
+/*
+ * Copyright @ 2019-present 8x8, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.videobridge.cc.av1
+
+import org.jitsi.nlj.PacketInfo
+import org.jitsi.nlj.RtpLayerDesc.Companion.getEidFromIndex
+import org.jitsi.nlj.rtp.codec.av1.Av1DDPacket
+import org.jitsi.nlj.rtp.codec.av1.Av1DDRtpLayerDesc
+import org.jitsi.nlj.rtp.codec.av1.Av1DDRtpLayerDesc.Companion.getDtFromIndex
+import org.jitsi.rtp.rtcp.RtcpSrPacket
+import org.jitsi.rtp.rtp.header_extensions.DTI
+import org.jitsi.rtp.util.RtpUtils
+import org.jitsi.rtp.util.isNewerThan
+import org.jitsi.utils.logging.DiagnosticContext
+import org.jitsi.utils.logging.TimeSeriesLogger
+import org.jitsi.utils.logging2.Logger
+import org.jitsi.utils.logging2.createChildLogger
+import org.jitsi.videobridge.cc.AdaptiveSourceProjectionContext
+import org.jitsi.videobridge.cc.RewriteException
+import org.jitsi.videobridge.cc.RtpState
+import org.json.simple.JSONArray
+import org.json.simple.JSONObject
+import java.time.Duration
+import java.time.Instant
+
+class Av1DDAdaptiveSourceProjectionContext(
+    private val diagnosticContext: DiagnosticContext,
+    rtpState: RtpState,
+    parentLogger: Logger
+) : AdaptiveSourceProjectionContext {
+    private val logger: Logger = createChildLogger(parentLogger)
+
+    /**
+     * A map that stores the per-encoding AV1 frame maps.
+     */
+    private val av1FrameMaps = HashMap<Long, Av1DDFrameMap>()
```

Comment:
Does this also need a limit?

Reply:
I don't think so, because only SSRCs that match a signaled encoding will get here.